interface/js_binding/ktx_wrapper.cpp

```diff
@@ -22,6 +49,136 @@ namespace ktx
         texture(texture&) = delete;
         texture(texture&& other) = default;
 
+        static texture createFromBuffer(const emscripten::val& data, int width, int height, int comps, bool srgb) {
```

Comment:
I'm now recalling some of the discussion about merging this work back to KTX-Software. This function meets the needs of the glTF-Compressor which only needs to make BasisU textures for KHR_texture_basisu but is far from complete as a general compression tool. That is why Phasmatic did not integrate it.

This function needs to have a format argument and many other parameters so it can create textures with various formats and shapes. Or perhaps there should be multiple functions, one for 2D textures like this is and additional functions for arrays, cube maps, 3d. We also need to expose the deflate functions.

We need to have a discussion.

Reply:
I want to capture this work in main as otherwise it will bit rot. I will review the changes again and mark what I think needs to be done to build only the ktx-read wrapper by default and to not release it the ktx wrapper.